### PR TITLE
[Chore] Update dependencies to use modern packages

### PR DESCRIPTION
**Pull request by Nava2**
This upgrades the plugin to use modern packages (e.g. generator version, assertj, etc.) and updates the build file to use kotlin. Unfortunately, as part of this, it was required to fix some bugs that **clearly** were never going to work as originally written. :smile:

Related to #17.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import com.gradle.publish.PublishTask.GRADLE_PUBLISH_KEY
```
Comment (Nava2) [inline build.gradle.kts:1]: This was as close to "ported" as I could do.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Fri May 26 20:19:53 EDT 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.1-bin.zip
```
Comment (Nava2) [inline gradle/wrapper/gradle-wrapper.properties:3]: @jamietanna / @scordio Gradle 7.6.1 seems reasonable 🤷🏻

File: gradlew [hunk 3]
```diff
@@ -1,78 +1,129 @@
-#!/usr/bin/env sh
+#!/bin/sh
```
Comment (Nava2) [inline gradlew:1]: generated

File: gradlew.bat [hunk 4]
```diff
@@ -1,84 +1,92 @@
-@if "%DEBUG%" == "" @echo off
```
Comment (Nava2) [inline gradlew.bat:1]: generated

**Comment by Nava2** [comment]
@jamietanna This still fails based on your work you were looking at. I wonder if we should ignore that `exclude_class` test to get this "version bump" commit in and then keep iterating on smaller changes.

File: src/test/groovy/org/assertj/generator/gradle/TestUtils.groovy [hunk 5]
```diff
@@ -34,11 +34,13 @@ class TestUtils {
                 mavenCentral()
             }
             
-            dependencies {                
+            dependencies {
+                implementation group: 'javax.annotation', name: 'javax.annotation-api', version: '1.3.2'
+
                 // https://mvnrepository.com/artifact/org.assertj/assertj-core
-                testCompile group: 'org.assertj', name: 'assertj-core', version: '3.8.0'
+                testImplementation group: 'org.assertj', name: 'assertj-core', version: '3.22.0'
```
Comment (scordio) [inline src/test/groovy/org/assertj/generator/gradle/TestUtils.groovy:41]:
```suggestion
                testImplementation group: 'org.assertj', name: 'assertj-core', version: '3.24.2'
```

File: src/test/groovy/org/assertj/generator/gradle/SimpleBuild.groovy [hunk 6]
```diff
@@ -116,14 +116,13 @@ class SimpleBuild {
...
-                compile group: 'com.google.guava', name: 'guava', version: '20.0'
+            dependencies { 
+                implementation group: 'javax.annotation', name: 'javax.annotation-api', version: '1.3.2'
                 
-                // https://mvnrepository.com/artifact/org.assertj/assertj-core
-                testCompile group: 'org.assertj', name: 'assertj-core', version: '3.8.0'
+                implementation group: 'com.google.guava', name: 'guava', version: '30.0-jre'
                 
-                testCompile group: 'junit', name: 'junit', version: '4.12'
+                testImplementation group: 'org.assertj', name: 'assertj-core', version: '3.22.0'
```
Comment (scordio) [inline src/test/groovy/org/assertj/generator/gradle/SimpleBuild.groovy:124]:
```suggestion
                testImplementation group: 'org.assertj', name: 'assertj-core', version: '3.24.2'
```

File: src/test/groovy/org/assertj/generator/gradle/SimpleBuild.groovy [hunk 7]
```diff
@@ -116,14 +116,13 @@ class SimpleBuild {
...
                         
-            dependencies {
-                // https://mvnrepository.com/artifact/com.google.guava/guava
-                compile group: 'com.google.guava', name: 'guava', version: '20.0'
+            dependencies { 
+                implementation group: 'javax.annotation', name: 'javax.annotation-api', version: '1.3.2'
                 
-                // https://mvnrepository.com/artifact/org.assertj/assertj-core
-                testCompile group: 'org.assertj', name: 'assertj-core', version: '3.8.0'
+                implementation group: 'com.google.guava', name: 'guava', version: '30.0-jre'
```
Comment (scordio) [inline src/test/groovy/org/assertj/generator/gradle/SimpleBuild.groovy:122]:
```suggestion
                implementation group: 'com.google.guava', name: 'guava', version: '31.1-jre'
```

**Review by scordio — COMMENTED** [review]
Thanks a lot, @Nava2. I just have a question about `burrunan/gradle-cache-action`, while the rest are mostly cosmetic comments that I can also apply on my own.

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - uses: actions/checkout@v3
+        with:
+          lfs: 'true'
+
+      - name: Verify gradle wrapper
+        uses: gradle/wrapper-validation-action@v1
```
[the remark this answers is not in the document]
Comment (Nava2) [reply .github/workflows/ci.yml:18]: I've got no preferences myself. I'll remove this.

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+      - name: Build plugin
+        uses: burrunan/gradle-cache-action@v1
+        with:
+          arguments: check --continue
```
[the remark this answers is not in the document]
Comment (Nava2) [reply .github/workflows/ci.yml:23]: Intention is that it avoids rebuilding, I copied this from one of my more ownerous builds. Lets use the generic one. 👍🏻

File: src/test/groovy/org/assertj/generator/gradle/SimpleBuild.groovy (hunk 6 above)
Comment (Nava2) [reply src/test/groovy/org/assertj/generator/gradle/SimpleBuild.groovy:124]: I intentionally matched the versions used in the generator itself: https://github.com/assertj/assertj-assertions-generator/blob/main/pom.xml

And proceeded to get them wrong, I've updated them accordingly though.

File: src/main/groovy/org/assertj/generator/gradle/internal/tasks/AssertionsGeneratorReport.groovy [hunk 10]
```diff
@@ -31,7 +32,7 @@ class AssertionsGeneratorReport {
     private Collection<String> inputPackages
     private Collection<String> inputClasses
     private Exception exception
-    private Collection<Class<?>> excludedClassesFromAssertionGeneration
+    private Collection<TypeToken<?>> excludedClassesFromAssertionGeneration
```
Comment (Nava2) [inline src/main/groovy/org/assertj/generator/gradle/internal/tasks/AssertionsGeneratorReport.groovy:35]: I have _no idea_ how this used to work. Groovy used to do some magic maybe?

File: src/test/groovy/org/assertj/generator/gradle/SimpleBuild.groovy [hunk 11]
```diff
@@ -140,6 +140,7 @@ class SimpleBuild {
     }
 
     @Test
+    @Ignore("Test fails due to groovy incompatibility")
```
Comment (Nava2) [inline src/test/groovy/org/assertj/generator/gradle/SimpleBuild.groovy:143]: I ignored this test for now, there's too much change here to understand why Groovy is going 💥

**Review by scordio — APPROVED** [review]
LGTM, although I cannot really judge all the changes 😅 @Nava2 I leave the merge to you, in case you spot something else that must be addressed before.

**Comment by Nava2** [comment]
> LGTM, although I cannot really judge all the changes 😅 @Nava2 I leave the merge to you, in case you spot something else that must be addressed before.

Since we aren't publishing anything, I think its safe to merge this and iterate.

**Pull request merged**